test_conformance/api/test_api_consistency.cpp

```diff
@@ -473,14 +437,689 @@ int test_consistency_pipes(cl_device_id deviceID, cl_context context,
                             sizeof(b), &b, NULL);
         test_error(error,
                    "Unable to query CL_DEVICE_GENERIC_ADDRESS_SPACE_SUPPORT");
-        if (b != CL_TRUE)
+        test_assert_error(
+            b == CL_TRUE,
+            "CL_DEVICE_PIPE_SUPPORT returned CL_TRUE but "
+            "CL_DEVICE_GENERIC_ADDRESS_SPACE_SUPPORT returned CL_FALSE");
+    }
+
+    return TEST_PASS;
+}
+
+int test_consistency_progvar(cl_device_id deviceID, cl_context context,
+                             cl_command_queue queue, int num_elements)
+{
+    // clGetDeviceInfo, passing CL_DEVICE_MAX_GLOBAL_VARIABLE_SIZE
+    // May return 0, indicating that device does not support Program Scope
+    // Global Variables.
+    cl_int error;
+
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+
+    size_t maxGlobalVariableSize = 0;
+    error = clGetDeviceInfo(deviceID, CL_DEVICE_MAX_GLOBAL_VARIABLE_SIZE,
+                            sizeof(maxGlobalVariableSize),
+                            &maxGlobalVariableSize, NULL);
+    test_error(error, "Unable to query CL_DEVICE_MAX_GLOBAL_VARIABLE_SIZE");
+
+    if (maxGlobalVariableSize == 0)
+    {
+        // Test setup:
+
+        error = create_single_kernel_helper(context, &program, &kernel, 1,
+                                            &test_kernel, "test");
+        test_error(error, "Unable to create test kernel");
+
+        size_t sz = 0;
```

Comment:
Probably best to be defensive and initialise the variable to a non-zero value.

Reply:
How strongly do you feel about this?  I'm sure this isn't the only test that initializes integral quantities to zero, and we'll return an error if the query fails so I think this is already fairly defensive.

Reply:
It's not the only one but this one is particularly bad since we're asserting that the value is 0 directly after the query. An incorrect implementation that doesn't return anything (doesn't write to memory) for `CL_DEVICE_GLOBAL_VARIABLE_PREFERRED_TOTAL_SIZE` will pass the check with `sz` initialised to 0. It wouldn't if `sz` was initialised to a different value. `SIZE_MAX`?

Reply:
I'll change the initialization value to `SIZE_MAX`.